src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/Debugging/TestLSPProjectionProvider.cs

```diff
@@ -3,47 +3,49 @@
 
 #nullable disable
 
-using System;
-using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.LanguageServer;
+using Microsoft.AspNetCore.Razor.LanguageServer.Protocol;
+using Microsoft.AspNetCore.Razor.Test.Common;
+using Microsoft.CodeAnalysis.Text;
 using Microsoft.VisualStudio.LanguageServer.ContainedLanguage;
 using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.LanguageServerClient.Razor.Extensions;
 using Microsoft.VisualStudio.LanguageServerClient.Razor.HtmlCSharp;
+using Microsoft.VisualStudio.LanguageServerClient.Razor.Test;
 
 namespace Microsoft.VisualStudio.LanguageServerClient.Razor.Debugging
 {
     internal class TestLSPProjectionProvider : LSPProjectionProvider
     {
-        private readonly Uri _documentUri;
-        private readonly IReadOnlyDictionary<Position, ProjectionResult> _mappings;
+        public static readonly TestLSPProjectionProvider Instance = new();
```

Comment:
Should add a private constructor to ensure folks use the class's `.Instance` property

Reply:
My question is why even bother with the instance property? In test code it seems unlikely that we need to avoid allocations, and there doesn't seem to be any shared caching going on.

Reply:
It's a fair question. I pesronally like the pattern from a readability and "don't inherit from this" perspective but you're right, it doesn't offer a tonnn of extra goods

Reply:
I like it from a readability standpoint too, for some reason it seems cleaner to me 😅